<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<html><head>

  
  <meta http-equiv="CONTENT-TYPE" content="text/html; charset=utf-8"><title>Yang Shen | Research</title>
  

  
  
  <meta name="GENERATOR" content="OpenOffice.org 2.0  (Linux)">

  
  <meta name="AUTHOR" content="Yang Shen">

  
  <meta name="CREATED" content="20080202;16374827">

  
  <meta name="CHANGED" content="20080206;18450100">

  
  <style type="text/css">
	<!--
		@page { size: 8.5in 11in }
		TD P { color: #000066 }
		H1 { color: #000066 }
		P { color: #000066 }
	-->
	</style>
  
  <script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-111381097-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

  </script>
  
  <script language="javascript">
  function toggleVis(divId){
	var arr = document.getElementById(divId)
	arr.style.display = (arr.style.display == 'none')? 'block':'none';
  }
  </script></head><body dir="ltr" style="color: rgb(0, 0, 102); background-color: rgb(255, 255, 255);" lang="en-US">
<dl>

  <dl>
    <dl>
      <dl>
        <dl>
          <dd>
            <table style="page-break-before: always; width: 680px; text-align: left; margin-left: auto; margin-right: auto;" border="0" cellpadding="2" cellspacing="0">
              <col width="104"> <col width="331"> <col width="165"> <col width="105"> <tbody>
                <tr>
                  <td style="width: 142px;" colspan="4"><br>
                  <h1 align="center"><small><font color="#000000"><font face="Arial, sans-serif">Yang Shen</font></font></small></h1>
                  <p style="margin-bottom: 0in;" align="center"><font color="#000000"><big><font face="Arial, sans-serif">[ </font><a href="index.html"><font face="Arial, sans-serif">Home</font></a> <font face="Arial, sans-serif">| <a href="research.html">Research</a> | </font></big></font><font color="#000000"><big><font face="Arial, sans-serif"><a href="publications.html">Publications</a>
|</font></big></font><font color="#000000"><big><font face="Arial, sans-serif"> <a href="capri.html">CAPRI</a> | </font><a href="cv.pdf"><font face="Arial, sans-serif">CV</font></a>
                  <font face="Arial, sans-serif">| </font><a href="links.html"><font face="Arial, sans-serif">Links</font></a> </big><font face="Arial, sans-serif"><big>]</big></font></font></p>
                  <span style="font-family: Arial,sans-serif;"><br>
                  </span>
                  <hr style="height: 1px;"> </td>
                </tr>
                <tr>
                  <td style="width: 142px;" colspan="4"> <br>
                  </td>
                </tr>
                <tr>
                  <td style="width: 142px; text-align: center;" colspan="4">
                  <div style="text-align: justify;"> </div>
                  <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;"><span style="text-decoration: underline;">Research Interests</span></span></font></p>
                  <ul style="text-align: left;">
<p style="text-align: left; color: blue;" class="MsoNormal"><span style="font-size: 16pt; line-height: 115%; font-family: &quot;Arial&quot;,&quot;sans-serif&quot;font-weight: bold;"> Our research theme is artificial intelligence (AI) for modeling biological intelligence (BI). Specifically, we develop optimization and machine learning among other AI algorithms for structural bioinformatics, structural systems biology, and biomedical and health informatics.  Some applications include </span></p>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;"><span style="text-decoration: underline;"></span>Protein or ligand design</span>
for binding affinity and specificity desired</font></li>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Structural
prediction</span> of protein interactions</font></li>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Systems
biology</span>, in particular, systems pharmacology</font></li>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Synthetic
biology</span>, with applications in energy and therapeutics</font></li>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Bioinformatics and big data</span></font></li>
                  </ul>
                  <div style="text-align: left;"><span style="font-weight: bold;">I am actively seeking experimental and
computational collaborations.</span>&nbsp; My main motivation is to
unravel molecular mechanisms and to modulate emergent behavior of
biomolecular networks with the development and application of
computational tools (including molecular modeling, network simulation,
optimization, machine learning, graph theory, and systems and control theory).&nbsp; To
that end, I aim at an iterative process that models and experiments can
feed each other. <br>
                  </div>
			  
		 <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Check out <a href="https://www.youtube.com/watch?v=5XD-WPpU1dE"><font face="Arial, sans-serif">a video about our research</font></a>. </font></p>	  
			  
                  <hr style="height: 3px;">
                  <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;"><span style="text-decoration: underline;">Group Members</span><br>
                  </span></font>
                  <ul style="text-align: left;">
                   <p style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Current members: </span></p>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Mostafa Karimi (Ph.D., 2015.9-)</font></li>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Yue Cao (Ph.D., 2016.9-)</font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Yuanfei Sun (Ph.D., 2018.6-)</font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Wuwei Tan (Ph.D., 2018.6-)</font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Di Wu (M.S./CE, 2017.9-2018.12; Research Associate 2019.2-)</font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Shaowen Zhu (M.S./ISS, 2018.3-2019.8; Ph.D. 2019.9-)</font></li>		    
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Fangtong Zhou (M.S./BIO, 2019.6-)</font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Yuning You (Ph.D. 2019.9-)</font></li>
		    
	            <p style="text-align: left; color: black;" class="MsoNormal"><span style=style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"> Alumni: </span></p>
		    <p>
	            Postdoc    
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Dr. Tomasz Oliwa (Postdoc 2013-14; First Employer: Univ. Chicago)</font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Dr. Su-Ping Deng (Postdoc 2018-19)</font></li>
		    </p>
		    <p>
		    MS	
	            <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Shaowen Zhu (M.S., Aug. 2019)
			    Thesis: <a href="">De Novo Protein Design of Novel Folds using Guided Conditional Wasserstein Generative Adversarial Networks (gcWGAN)</a></font>
	            <DT>Next: Ph.D. Program in EE at TAMU
		    </li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Di Wu (M.S., Dec. 2018)
			    Thesis: <a href="http://hdl.handle.net/1969.1/174625">Structured Sparsity Learning for Coevolution-Based Protein Contact Prediction</a></font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Haoran Chen (M.S., Aug. 2018)
			    Thesis: <a href="http://hdl.handle.net/1969.1/174024">Improving molecular-level protein docking and interpreting system-level cancer mechanism through machine learning</a></font>
			    <DT>Next: Ph.D. Program in Comp Biol at CMU 
	            </li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Yi Yan (M.S., May 2018)
			    Thesis: <a href="http://hdl.handle.net/1969.1/173628">Efficiency prediction and mechanism discovery for the CRISPR-Cas9 system</a></font></li>    
		    </p>
		    <p>
	            ME
	 	    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Tianliang Li (Member 2016-18; M.Eng. 2018)</font></li>
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Yuanfei Sun (Grad Summer Intern, 2016; M. Eng. 2017; Next: Ph.D. in Electrical Engineering at TAMU)</font></li>
                    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Kuei-Fu Chen (Grad Summer Intern, 2016; M.Eng. 2017; First Employer: Bloomberg L.P.)</font></li>
		    </p>	    
		    <p>	 
	            Undergraduate
	 	    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Muhammad Ahmad (ECEN 491 Research Mentee, Fall 2019)</font></li>	
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">S. S. N. Vishnu Siva Sai (Research Exchange Undergrad (REU) from IIT-Hyderabad, Summer 2019)</font></li>			
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Annie Rizvi (Research Intern, Summer 2018)</font></li>			    
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Oluwaseyi Moronfoye (Research Intern, Spring 2018; Undergrad Summer Research Grant (USRG), Summer 2018; Next: M.S. in Finance at Mays Business School at TAMU)</font></li>	    
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Elsherif Mahmoud (Research Exchange Undergrad (REU) from TAMU-Qatar, Summer 2017; Next: M.S. in Electrical Engineering at Columbia University)</font></li>			  
		    <li><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Clarissa Tovias (Research Intern, Summer 2016)</font></li>
		    </p>
			  
                  </ul>	  
	          </p>
			  
			  
                  <hr style="height: 3px;">
                  <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;"><span style="text-decoration: underline;">Current Projects</span><br>
                  </span></font></p>

                  <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Dimension Reduction and Optimization Methods for Flexible Refinement of Protein Docking (NSF <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1546278">CCF-1546278</a>)<br>


                  <hr style="height: 3px;">
                  <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;"><span style="text-decoration: underline;">Past Projects</span><br>
                  </span></font></p>
                  <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Design for Promiscuous
Inhibitors<br>
                  </span></font></p>
                  <p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">Keywords: Structure-based drug design, binding affinity and
specificity, drug resistance, combinatorial optimization, machine
learning</font></p>
                  <p style="text-align: left; color: black;" class="MsoNormal"><span style="font-size: 12pt; line-height: 115%; font-family: &quot;Arial&quot;,&quot;sans-serif&quot;;">In
this study we aim
to understand molecular mechanisms by which small molecules can exhibit
binding
promiscuity and to develop design strategies to implement such
promiscuity.&nbsp; We
chose to study inhibition of HIV-1 protease because robust HIV-1
protease
inhibition in the face of an evolving viral population remains a
tremendous
challenge, yet significant structural, inhibitory, and evolutionary
data exist
upon which to base our investigation.&nbsp; Using computational design
we
constructed
small-molecule inhibitors targeting a set of wild-type and
drug-resistant
mutant HIV-1 proteases.&nbsp; The resulting inhibitor library contained
rich
diversity
in term of chemical structures and binding specificity profiles.&nbsp;
Subsequent statistical
analysis of this large library of inhibitors revealed significant
trends for
promiscuous inhibitors targeting multiple variants of HIV-1 protease. <span style="">&nbsp;</span>Accordingly, we came up with potential
design strategies for promiscuous HIV-1 protease inhibitors.</span></p>
                  <p style="text-align: left; color: black;" class="MsoNormal"><span style="font-size: 12pt; line-height: 115%; font-family: &quot;Arial&quot;,&quot;sans-serif&quot;;"><br>
                  </span></p>
                  <div style="text-align: left;"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Charge Optimization Theory for Flexible Ligands<br>
                  <br>
                  </span>Keywords: lead optimization, electrostatic
optimization, conformational flexibility, continuum electrostatics,
linear response theory, linearized Poisson–Boltzmann equation<br>
                  <br>
The design of ligands with high affinity and specificity remains a
fundamental challenge in understanding molecular recognition and
developing therapeutic interventions.&nbsp; Charge optimization theory
addresses this problem by determining ligand charge distributions
producing the most favorable electrostatic contribution to the binding
free energy.&nbsp; The&nbsp; theory has been applied to design of
binding specificity as well.&nbsp; However, the&nbsp; formulations
described only treat a rigid ligand. Here we extend the theory to
treat&nbsp; flexible ligands.&nbsp; We develop a thermodynamic pathway
analysis for binding contributions relevant to the theory, and we
illustrate application of the theory using HIV-1 protease with our
previously designed and validated subnanomolar inhibitor.&nbsp; The
results show that flexible ligand binding is not an adaptation designed
to enhance binding affinity, and lead to new insights into the sources
of comformational changes that accompany binding.<br>
                  <span style="font-weight: bold;"><br>
                  <br>
                  </span></font></div>
                  <div style="text-align: left;"><span style="font-family: Arial,sans-serif;"><span style="font-weight: bold;"></span></span><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Protein Docking <br>
by Exploring and Exploiting the Binding Free Energy Landscape<br>
                  </span><br>
Keywords: Structure prediction, </font><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">binding free energy
landscape, </font><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">binding funnel, global optimization,
semi-definite programming, special Euclidean group, Riemannian manifold, exponential
coordinates<br>
                  <br>
Protein</font><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">–</font><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">protein interactions play a central role in various aspects
of the structural and functional organization of the cell.&nbsp;
Protein docking, the computational prediction of such interactions at
the atomic level, is crucial for better understanding cellular
processes and provides valuable information for rational drug
design.&nbsp; Similarly to protein folding, the association of two
proteins is driven by a free energy funnel, determined by favorable
interactions in some neighborhood of the native state.&nbsp; We develop
a docking method based on stochastic global minimization of
funnel-shaped energy functions in the space of rigid body motions
(SE(3), a Riemannian manifold) while accounting for flexibility of the interface side
chains.&nbsp; The method, called semi-definite programming-based
underestimation (SDU), employs a general quadratic function to
underestimate a set of local energy minima and uses the resulting
underestimator to bias further sampling. <br>
                  <br>
While SDU effectively minimizes functions with funnel-shaped basins,
its application to docking in the rotational and translational space
SE(3) is not straightforward due to the geometry of that space (a manifold).&nbsp;
We introduce a strategy that uses separate independent variables for
side-chain optimization, center-to-center distance of the two proteins,
and five angular descriptors of the relative orientations of the
molecules that approximate $S^2 \times SO(3)$ with a Euclidean space.&nbsp; This algorithm explores the free energy surface
spanned by encounter complexes that correspond to local free energy
minima and shows similarity to the model of macromolecular association
that proceeds through a series of collisions.&nbsp; Results for
standard protein docking benchmarks establish that in this space the
free energy landscape is a funnel in a reasonably broad neighborhood of
the native state and that the SDU strategy can generate good docking
predictions (interface RMSD &lt;= 5Å) while achieving at least 20-fold
efficiency gain compared to Monte Carlo methods.<br>
                  <br>
I have been continuously developing and applying SDU in CAPRI, a
community-wide assessment of protein docking methods, with considerable
success. (<a href="capri.html">More details</a>)<br>
                  <br>
                  <br style="font-weight: bold;">
                  <span style="font-weight: bold;">Metabolic Engineering<br>
                  <br>
                  </span>Keywords: Microbial fuel cell design, systems
and synthetic biology, flux balance analysis, bi-level optimization,
relaxations<br>
                  <br>
Biological networks are abstract representations of biological systems
that capture many essential characteristics of other networks.&nbsp;
Molecules in a living cell are nodes of such networks and their
interactions form undirected edges or directed arcs.&nbsp; Biological
systems are proved to share structural principles with engineered
networks.&nbsp; The engineering analysis of biological networks help
understand a cell's functional organization and manipulate
biotransformations in these networks. Such biotransformations aim at
various biochemical by-products varying from simple chemicals to even
microbes. In this collaboration with synthetic biologists at BU, we
develop computational methods to predict environmental conditions and
genetic modulations to optimize the production efficiency / yield of
metabolic by-products (electrons), based on flux balance analysis.<br>
                  <span style="font-weight: bold;"></span></font><br>
                  <br>
                  <font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><span style="font-weight: bold;">Binding
Site Similarity of Analogous Enzymes<br>
                  <br>
                  </span>Keywords: Structure-function analysis,
convergent evolution, binding site, fast Fourier transform<br>
                  <br>
Most enzymes catalyze specific biochemical reactions of well-defined
substrates.&nbsp; Both the catalysis and recognition usually occur in
localized cavities or binding sites.&nbsp; The general goal of this
study is to characterize the similarities and differences in local
sequence and structure of the binding sites of enzymes that perform the
same or similar chemical reactions.&nbsp; The comparisons are based on
the optimal structural superimpositions of the binding sites rather
than the global sequence or structure similarity of the enzymes. <br>
                  <span style="color: rgb(0, 0, 0); font-family: Arial; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; line-height: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; background-color: rgb(255, 255, 255); font-size: small; display: inline ! important; float: none;"><br></span><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">We study analogous enzymes that have the same enzyme commission (EC) number (or whose EC
numbers differ only in the last digit), but are evolutionarily
unrelated, i.e., they lack both sequence and structural similarity.
Analogous enzyme pairs are relatively rare, but occur in all major
classes, assumed to be the results of convergent evolution.&nbsp; Research on
analogous enzymes is very limited: it consists of searches for
non-homologous enzymes with the same EC number and studies of specific
cases of convergent evolution.&nbsp; It is known that at least in a number of
cases the spatial arrangement of the catalytic residues is conserved,
but very little is known about the similarity of the binding sites that
occur on different protein scaffolds.&nbsp; In this work we use a new method
developed to assess molecular similarity for the structural
superimposition of enzyme binding sites.&nbsp; The physicochemical properties
of the cavity-flanking residues are represented by pseudocenters. Given
two sets of such pseudocenters, our goal is finding the largest subset
of pseudocenters in both clefts in direct correspondence with each
other geometrically as well as chemically. The new method performs
an exhaustive evaluation of the correlation function in the discretized
6D space of mutual orientations of the two point sets using a very
efficient algorithm involving fast Fourier transforms. The method is
applied to a number of analogous enzyme pairs and provides new insights into molecular mechanisms of their functions. <span style="color: rgb(0, 0, 0); font-family: Arial; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; line-height: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; background-color: rgb(255, 255, 255); font-size: small; display: inline ! important; float: none;"><br>
                  <br>
                  </span><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">I
was also involved in small-molecule solvent mapping study and its
applications to hot spot prediction and fragment-based drug design.</font><br>
                  </font></font></div>
<font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">                  </font></font><p style="text-align: left;" class="MsoNormal"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><o:p></o:p></font></font></font></p>
<font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3"><font style="font-size: 12pt;" color="#000000" face="Arial, sans-serif" size="3">                  </font></font></td>
                </tr>
                <tr>
                  <td style="width: 142px;" colspan="4"><br>
                  </td>
                </tr>
              </tbody>
            </table>
          </dd>
        </dl>
      </dl>
    </dl>
  </dl>
</dl>

<script>
/*
Message Slideshow
By Website Abstraction (www.wsabstract.com)
and Java-scripts.net (www.java-scripts.net)
Credits must stay intact for use
*/

var curmsg=-1
var messages=new Array()
messages[1]="Yang is now on the job market!"
messages[0]="Oral presentation ``Designing promiscuous inhibitors: lessons learned from HIV-1 protease inhibition'' accepted by ACS."
messages[2]="Check Yang's shared list of readings."
//add more messages as desired

var messagelinks=new Array()
messagelinks[1]="http://engineering.academickeys.com/seeker_job.php?dothis=select&form%5Bcategory_IDXs%5D%5B%5D=4&form%5Bfield_IDXs%5D%5B%5D=3&job%5Bctry%5D=&job%5Binst_name%5D=&job%5Bdescrip%5D=&start=0"
messagelinks[0]="http://portal.acs.org/portal/PublicWebSite/meetings/spring2011/index.htm"
messagelinks[2]="http://www.google.com/reader/shared/user/05673326590867862879/state/com.google/broadcast"
//add more links as indicated by the number of messages 

function slidemessage(){
if (curmsg<messages.length-1)
curmsg++
else
curmsg=0
document.slideshow[0].value=messages[curmsg]
setTimeout("slidemessage()",3000)
}
slidemessage()
</script>
</body></html>
